primitives/src/channel_validator.rs

```diff
@@ -45,7 +50,9 @@ pub fn all_validators_listed(validators: &SpecValidators, whitelist: &[String])
     } else {
         let found_validators = whitelist
             .iter()
-            .filter(|&allowed| {
+            .map(|identity| ValidatorId::try_from(&identity[..]))
```

Comment:
Shouldn't the `whitelist` also be `ValidatorId` slice?
PS: Also this `&identity[..]` here again.